examples/diff.c

```diff
@@ -335,3 +357,47 @@ static void diff_print_stats(git_diff *diff, struct opts *o)
 	git_buf_free(&b);
 	git_diff_stats_free(stats);
 }
+
+struct pair extract_from_string(const char *cmd_opt)
+{
+	struct pair p;
+	char **endptr;
+
+
+	if(*cmd_opt=='/') {
+		p.first = DEFAULT_THRESHOLD;
+		cmd_opt++;
+		p.second = strtol(cmd_opt,endptr,0);
+		if(**endptr!='%') giterr_set_str(GITERR_INVALID, "Invalid input format.");
+	} else {
+		p.second = DEFAULT_BREAK_REWRITE_THRESHOLD;
+		p.first = strtol(cmd_opt,endptr,0);
+		if(**endptr!='%') giterr_set_str(GITERR_INVALID, "Invalid input format.");
```

Comment:
`giterr_set_str()` is for setting the library's error for this thread; a consumer of the library (which all of these examples are) should not be using it to set its error. I also don't see how this function is signaling an error. How does a caller of `extract_from_string()` know that the option was invalid? `extract_m_n()` doesn't seem to care whether there was an error.

The program should either die if there are errors parsing the command or this function has to signal an error to the caller.

This isn't checking whether `strtol` reported an error. Passing just `%` as the option would make this code think that I passed in a valid number 0, which is not true. AFAICT git also restricts this to base-10 numbers, so we shouldn't be letting the user of this diff program choose to use octal or hex for their percentages.

The code is also not formatted according to the rules we follow in the codebase. Don't put the code inside the if on the same line and leave a space between `if` and the parenthesis.